spi/src/transformations/project.ts

```diff
@@ -29,24 +29,65 @@ export const transformProjects = async (connection: IDBConnection): Promise<void
             SELECT 
                 p.spi_project_id, 
                 pp.person_id, 
-                b.project_id
+                b.project_id, 
+                CONCAT(pp.first_given_name, ' ', pp.surname) AS full_name,
+                spp.email_address
             FROM 
                 public.spi_projects p
-            INNER JOIN 
-                biohub.project b ON p.spi_project_id = b.spi_project_id
-            INNER JOIN 
-                spi_persons pp ON pp.spi_project_id = p.spi_project_id
+            INNER JOIN biohub.project b 
+                ON p.spi_project_id = b.spi_project_id
+            INNER JOIN spi_persons pp 
+                ON pp.spi_project_id = p.spi_project_id
+            LEFT JOIN public.spi_secure_persons spp 
+                ON spp.first_name = pp.first_given_name
+                AND spp.last_name = pp.surname
         )
     INSERT INTO
         biohub.project_participation (project_id, system_user_id, project_role_id, create_user)
     SELECT DISTINCT
         w_mapping.project_id,
-        (SELECT biohub_user_id FROM public.migrate_spi_user_deduplication WHERE w_mapping.person_id = ANY (spi_person_ids)),
-        (SELECT project_role_id FROM biohub.project_role WHERE name = 'Collaborator'),
-        (SELECT system_user_id FROM biohub.system_user WHERE user_identifier = 'spi')
+
+        COALESCE (
+        (SELECT biohub_user_id 
+         FROM public.migrate_spi_user_deduplication 
+         WHERE w_mapping.person_id = ANY (spi_person_ids)
+        ), 
+        (SELECT system_user_id 
+         FROM biohub.system_user 
+         WHERE user_identifier = 'spi')
+        ) AS system_user_id,
+
+        CASE 
+        WHEN p.coordinator IS NOT NULL AND w_mapping.full_name LIKE '%' || TRIM(BOTH ', ' FROM p.coordinator) || '%' AND spp.email_address LIKE '%@gov.bc.ca%' THEN 
```

Comment:
```CASE
    WHEN p.coordinator = w_mapping.full_name AND spp.email_address LIKE '%@gov.bc.ca%' THEN (SELECT project_role_id FROM biohub.project_role WHERE name = 'Coordinator'
  WHEN spp.email_address LIKE '%@gov.bc.ca%' THEN (SELECT project_role_id FROM biohub.project_role WHERE name = 'Collaborator'
  ELSE (SELECT project_role_id FROM biohub.project_role WHERE name = 'Observer')
```

Reply:
simplified in code! Would this still ensure that when no coordinator is set, that the spi user is set as coordinator?